packages/react-query/src/useMutationState.ts

```diff
@@ -47,31 +47,26 @@ export function useMutationState<TResult = MutationState>(
 ): Array<TResult> {
   const mutationCache = useQueryClient(queryClient).getMutationCache()
   const optionsRef = React.useRef(options)
-  const result = React.useRef<Array<TResult>>(null)
-  if (!result.current) {
-    result.current = getResult(mutationCache, options)
-  }
-
   React.useEffect(() => {
     optionsRef.current = options
   })
 
-  return React.useSyncExternalStore(
-    React.useCallback(
-      (onStoreChange) =>
-        mutationCache.subscribe(() => {
-          const nextResult = replaceEqualDeep(
-            result.current,
-            getResult(mutationCache, optionsRef.current),
-          )
-          if (result.current !== nextResult) {
-            result.current = nextResult
-            notifyManager.schedule(onStoreChange)
-          }
-        }),
-      [mutationCache],
-    ),
-    () => result.current,
-    () => result.current,
-  )!
+  const [result, setResult] = React.useState<Array<TResult>>(() =>
+    getResult(mutationCache, options),
+  )
+
+  React.useEffect(() => {
+    const sub = mutationCache.subscribe(() => {
```

Comment:
one thing that's missing here is calling the setter directly in useEffect, like we do in  `useBaseQuery` and `useIsFetching` to make sure we do not miss any query updates between creating the observer and subscribing to it.

I know it's not really possible because the result isn't necessarily a primitive.

Do you think it would be possible to keep `useSyncExternalStore` for `useMutationState` and `useIsFetching` ?

Reply:
I'm not sure if we should keep it - it seems to me like something that should be deprecated in React if it by design causes issues as soon as you use `startTransition`?

Reply:
Or maybe it does make sense to use uSES here, idk....

Reply:
Reveted them too, 76bfacf for future reference